src/troute-network/troute/DataAssimilation.py

```diff
@@ -115,4 +386,438 @@ def last_obs(self):
 
     @property
     def usgs_df(self):
-        return self._usgs_df
\ No newline at end of file
+        return self._usgs_df
+
+    
+class AllDA(DataAssimilation):
+    """
+    Object containing all data assimilation information. Created from user input
+    and a network object that defines the network and forcing parameters. TODO: this
+    could be broken up into separate objects specific to each type of DA (streamflow
+    nudging, reservoir persistence, etc.). But for now it is all done here.
+    
+    Arguments:
+    ----------
+    - data_assimilation_parameters (dict): user input data re data assimilation
+    - run_parameters               (dict): user input data re subset of compute configuration
+    - waterbody_parameters         (dict): user input data re waterbody parameters
+    - network                    (Object): network object created from abstract class
+    - da_run                       (list): list of data assimilation files separated
+                                           by for loop chunks
+    
+    Returns:
+    --------
+    - data_assimilation               (Object): Object containing all data assimilation information
+        - usgs_df                  (DataFrame): dataframe of USGS gage observations
+        - lastobs_df               (DataFrame): Last gage observations data for DA
+        - reservoir_usgs_df        (DataFrame): USGS reservoir observations
+        - reservoir_usgs_param_df  (DataFrame): USGS reservoir DA parameters
+        - reservoir_usace_df       (DataFrame): USACE reservoir observations
+        - reservoir_usace_param_df (DataFrame): USACE reservoir DA parameters
+        - da_parameter_dict             (dict): user input data re data assimilation
+        - waterbody_types_df       (DataFrame): Waterbody type codes (1:levelpool, 2:USGS, 3:USACE, 4:RFC)
+        - usgs_lake_gage_crosswalk      (dict): usgs gage ids and corresponding segment 
+                                                ids at which they are located
+        - usace_lake_gage_crossalk      (dict): usace gage ids and corresponding segment 
+                                                ids at which they are located
+    """
+    __slots__ = ["_usgs_df", "_last_obs_df", "_reservoir_usgs_df", "_reservoir_usgs_param_df", "_reservoir_usace_df", "_reservoir_usace_param_df", "_da_parameter_dict", "_waterbody_types_df", "_usgs_lake_gage_crosswalk", "_usace_lake_gage_crosswalk"]
+    def __init__(self, data_assimilation_parameters, run_parameters, waterbody_parameters, network, da_run):
+        lastobs_df, da_parameter_dict = nnu.build_data_assimilation_lastobs(
```

Comment:
why isn't this `build_data_assimilation_lastobs` functions moved to this module?

Reply:
No reason. I've now included this function within this module.